.github/workflows/luacheck.yml

```diff
@@ -0,0 +1,23 @@
+name: Luacheck
```

Comment:
instead of adding github actions and yet more config to maintain, can't we add it as a separate job to Travis?

Reply:
I picked actions for this because it offers several things Travis doesn't make easy. I've been down the road of trying to keep all CI jobs under one platform, but have decided it's a terrible trade off. You end up fighting it to get it to do things it isn't good at and piling up hacks in the name of maintainability.

We already have 2 (Travis & Appveyor) and this makes perfect sense, on is running Linux builds against a matrix, the other Windows. Use the best tool for the job.

In the case of linting, it's not really the same kind of check as the regression suite and doesn't need to be inside a matrix. Doing both matrix and non-matrix jobs in a Travis config is a lot messier than having a separate workflow, and even when you get it working you can't do what this will do: add a separate line item to PR's with exactly what failed. We can even configure this quite easily to add PR comments to changed line numbers when something goes wrong (I didn't set that up but it's a logical next step once the linter is running on Actions).

Reply:
I get the reasoning, but to me it is also a matter of volume. If it is a project with new PR's every day, I'd immediately go for this change. But since this is a low volume project, the maintenance of more tools is going to be a relatively bigger part of the overall effort.

Reply:
Since you're firmly into this, let's go for it.

Reply:
Thanks. One final thought on the 'maintenance' issue: having different tasks/jobs in different runners (no matter what the CI platform used) reduces maintenance because it's easier to copy/paste code. I know this Luacheck setup works because I use it almost bit-for-bit the way it appears here on several other projects. On the other hand Travis's one-runner per project approach means adding or removing jobs takes careful consideration per-project on how to fit that task into the overall runner configuration. As a separate runner for a specific job you can add/remove/change/enable/disable/update a single check without any risk of unintended consequences to an some other set of elaborate jobs.